src/services/impl/FileTextReader.hh

```diff
@@ -51,13 +51,13 @@ public:
             }
         }
 
-        return static_cast<size_t>(const_cast<std::ifstream&>(m_iStream).tellg());
+        return static_cast<size_t>(m_iStream.tellg());
```

Comment:
I made the return/parameter types `std::streampos` (since they are supposedly positions), for data members I put `std::streamoff` to keep classes nothrow `DefaultConstructible`. They are implicitly convertible to each other. `std::streamoff` is a `typedef` of `long long`. `std::streampos` keeps track of the state.